### PR TITLE
M3-4961 Change: Separate password change flow for bare metal Linodes

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Bare metal Linodes don't have disks that are managed through the API, so they have to use a new endpoint (linode/instances/:id/password) to reset their root password. In planning I recommended using a separate component for this to avoid a bunch of prop-driven stuff, but it turns out there's a lot of shared logic between the two flows that would be repeated and I couldn't think of a great way around it. So I kept the existing stuff in place and a) hid the disks dropdown then b) used a different method on submit.

### CODE REVIEW

File: packages/manager/src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsPasswordPanel.tsx [hunk 1]
```diff
@@ -73,27 +74,34 @@ class LinodeSettingsPasswordPanel extends React.Component<
...
+          set(lensPath(['errors']), errors),
+          set(lensPath(['submitting']), false)
+        ),
+        () => {
+          scrollErrorIntoView('linode-settings-password');
+        }
+      );
+    };
+
+    if (isBareMetalInstance) {
```
Comment (johnwcallahan) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeSettings/LinodeSettingsPasswordPanel.tsx:99]: Missing resetLinodePassword call here?

**Review by johnwcallahan — COMMENTED** [review]
I'm unable to test this right now since I don't have any bare metal instances, but the code looks OK aside from one thing.

**Review by tiffwong — APPROVED** [review]
Tested with mocks, everything looks good

File: packages/api-v4/src/linodes/linodes.ts [hunk 2]
```diff
@@ -117,3 +117,20 @@ export const deleteLinode = (linodeId: number) =>
...
+ * rather than on a specific disk. This situation only applies
+ * to bare metal instances, which don't have disks that are managed
+ * through the API.
+ */
+
+export const changeLinodePassword = (linodeId: number, root_pass: string) =>
+  Request<{}>(
+    setURL(`${API_ROOT}/linode/instances/${linodeId}/password`),
+    setData({ root_pass }),
+    setMethod('DELETE')
```
Comment (DevDW) [inline packages/api-v4/src/linodes/linodes.ts:135]: Should this be a `POST`?

File: packages/api-v4/src/linodes/linodes.ts [hunk 3]
```diff
@@ -117,3 +117,20 @@ export const deleteLinode = (linodeId: number) =>
     setURL(`${API_ROOT}/linode/instances/${linodeId}`),
     setMethod('DELETE')
   );
+
+/**
+ * resetLinodePassword
+ *
+ * This method is distinct from resetLinodeDiskPassword,
```
Comment (DevDW) [inline packages/api-v4/src/linodes/linodes.ts:124]: `changeLinodeDiskPassword` ?

**Review by DevDW — APPROVED** [review]
LGTM aside from a couple of comments

**Pull request merged**